### PR TITLE
Feat snowflake state handler

### PR DESCRIPTION
**Pull request by Matt9993**
## Summary
This pull request contains a feature that leverages the already existing state handler functionality of e.g. slack notifications but instead of a slack message or email this specific state handler acts as a custom logger that stores the Task information in a Data Warehouse table.

```python
        from prefect import task
        from prefect.utilities.notifications import snowflake_logger
        @task(state_handlers=[snowflake_logger(ignore_states=[Running])])
        def add(x, y):
            return x + y
 ```

## Changes
This PR doesn`t change anything although adds a new state handler / logger option

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by madkinsz** [comment]
@Matt9993 Thanks for contributing this! I'm going to request some reviewers that are more relevant :)

**Review by dylanbhughes — APPROVED** [review]
LGTM!

File: src/prefect/utilities/notifications/notifications.py [hunk 1]
```diff
@@ -318,3 +318,136 @@ def add(x, y):
...
+    test_env: bool = False,
+) -> "prefect.engine.state.State":
+    """
+    Snowflake state change handler/logger; requires having the Prefect Snowflake app installed.
+    Works as a standalone state handler, or can be called from within a custom state handler.  This
+    function is curried meaning that it can be called multiple times to partially bind any
+    keyword arguments (see example below).
+    Args:
+        - tracked_obj (Task or Flow): Task or Flow object the handler is
+            registered with
```
Comment (tpdorsey) [inline src/prefect/utilities/notifications/notifications.py:364]:
```suggestion
            registered with.
```

File: src/prefect/utilities/notifications/notifications.py [hunk 2]
```diff
@@ -318,3 +318,136 @@ def add(x, y):
...
+        - old_state (State): previous state of tracked object
```
Comment (tpdorsey) [inline src/prefect/utilities/notifications/notifications.py:365]:
```suggestion
        - old_state (State): Previous state of tracked object.
```

Comment (tpdorsey) [reply src/prefect/utilities/notifications/notifications.py:365]: Docstrings should be sentences, starting capital and closing period. Examples above.

**Comment by madkinsz** [comment]
@ahuang11 can we get this across the line?

**Comment by ahuang11** [comment]
Not sure why the checks are failing.
```
Run actions/labeler@v4
Error: HttpError: Resource not accessible by integration
Error: Resource not accessible by integration
```

```

2:17:03 PM: /opt/build-bin/run-build-functions.sh: line 223: /opt/buildhome/python3.7/bin/activate: No such file or directory
2:17:03 PM: Error setting python version from runtime.txt
2:17:03 PM: Please see https://github.com/netlify/build-image/blob/focal/included_software.md for current versions
2:17:03 PM: Build was terminated: Build script returned non-zero exit code: 1
```

Don't think it's related to the PR

**Review by desertaxle — APPROVED** [review]

**Pull request merged**